### DISCUSSION TITLE
Would love to see a count of the total Tailwind classes being used

### DISCUSSION THREAD

**Discussion by silouanwright** [tailwindlabs/tailwindcss#9720] [Ideas]
Hello,

One of the things I'm trying to do in my organization is drum up excitement for how much more we're using Tailwind.

I would simply love to figure out, how many Tailwind classes we're using altogether. For my specific use case, I'm looking for all the Tailwind classes I'm using (even repeats). I think it would be an amazing feature to simply output the number of scanned classes at the end of the tailwind build command or to add a flag to make it optional.

In the meantime, if anyone has any ideas on a way to do this manually, please let me know!

**Comment by wongjn** [comment]
You could use PostCSS to parse the generated CSS into an abstract syntax tree and count the number of rules that have class selectors.

**Reply by silouanwright** [reply]
Good idea! But parsing the CSS itself would just give me unique classes right? I want the repeats as well. Meaning, if `text-center` is used 10 times in my entire application, I want to see the number 10. I would need to parse the scanned files themselves it seems like. I could theoretically parse the CSS, assemble a giant regex of classes and then search my repo with that? But that would be error prone.

But I really feel like it would be a nice win to provide this functionality in the CLI, as it seems like many tools output statistics of what was processed after the commands runs. And seeing the number of Tailwind classes would encourage projects not fully migrated to utilize it more.
